test-data/unit/check-flags.test

```diff
@@ -1232,3 +1232,37 @@ A = List  # OK
 B = List[A]  # E:10: Missing type parameters for generic type "A"
 x: A  # E:4: Missing type parameters for generic type "A"
 [builtins fixtures/list.pyi]
+
+[case testIgnoreErrorCodesGlobal]
+# flags: --ignore-error-codes=attr-defined,arg-type
+
+not_defined  # E: Name 'not_defined' is not defined
+x: int = 'no'  # E: Incompatible types in assignment (expression has type "str", variable has type "int")
+'no'.no_way
+def test(x: str) -> None: ...
+test(0)
+
+[case testIgnoreErrorCodesPerFile]
+# flags: --config-file tmp/mypy.ini
+import b
+not_defined
+x: int = 'no'
+'no'.no_way
+def test(x: str) -> None: ...
+test(0)
+[file b.py]
+not_defined
+x: int = 'no'
+'no'.no_way
+def test(x: str) -> None: ...
+test(0)
+[file mypy.ini]
+\[mypy]
+ignore_error_codes = attr-defined, arg-type
+\[mypy-b]
+ignore_error_codes = name-defined, assignment
```

Comment:
I'd like to eventually move various strictness flags to be enabled/disabled through error codes. To do this, it seems that per-file options need to be a delta over global settings, instead of replacing them. Also, command-line options probably should be a delta over settings in the ini file. I.e., the ignored error codes would be the union of error codes ignored in the config file, on the command line, and in per-file options. 

The logic gets a bit more involved once we allow enabling error codes as well, but at least we should be able to design the semantic for ignoring error codes in a way that we don't need to introduce a compatibility break.